src/HaskellCI/GitHub/Yaml.hs

```diff
@@ -30,7 +30,7 @@ data GitHubJob = GitHubJob
     { ghjName      :: String
     , ghjRunsOn    :: String
     , ghjContainer :: Maybe String
-    , ghjMatrix    :: [M.Map String String]
+    , ghjMatrix    :: [M.Map String (Yaml [String])]
```

Comment:
These types shouldn't reference `Yaml`.

Reply:
What is the alternative? `ghc` needs to map to a `String`, while `continue-on-error` needs to map to a `Bool`.

Reply:
new datatype!

Reply:
I have added a new `GitHubMatrixValue` with two constructors, one for `String` and one for `Bool`. Is this what you had in mind?